src/transformers/generation/watermarking.py

```diff
@@ -0,0 +1,238 @@
+# coding=utf-8
+# Copyright 2024 The HuggingFace Inc. team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+from dataclasses import dataclass
+from functools import lru_cache
+from typing import Dict, Optional, Union
+
+import numpy as np
+
+from ..configuration_utils import PretrainedConfig
+from ..utils import is_torch_available, logging
+from .configuration_utils import WatermarkingConfig
+
+
+if is_torch_available():
+    import torch
+
+    from .logits_process import WatermarkLogitsProcessor
+
+
+logger = logging.get_logger(__name__)
+
+
+@dataclass
+class WatermarkDetectorOutput:
+    """
+    Outputs of a watermark detector.
+
+    Args:
+        num_tokens_scored (np.array of shape (batch_size)):
+            Array containing the number of tokens scored for each batch.
+        num_green_tokens (np.array of shape (batch_size)):
+            Array containing the number of green tokens for each batch.
+        green_fraction (np.array of shape (batch_size)):
+            Array containing the fraction of green tokens for each batch.
+        z_score (np.array of shape (batch_size)):
+            Array containing the z-score for each batch.
```

Comment:
is there a more reprensentative name for that?

Reply:
It is a term from stats, so I guess we cannot just call it how we want. I added a small explanation of what is z-score